### PR TITLE
Pass helper to ha-form subtypes

### LINKED ISSUE #14164: Non selectors can not use data_description in configuration flow.
### Checklist

- [X] I have updated to the latest available Home Assistant version.
- [X] I have cleared the cache of my browser.
- [X] I have tried a different browser to see if it is related to my browser.

### Describe the issue you are experiencing

Descriptions for fields during configuration flow for Integrations not using data selectors, but standard validators like cv.string, are not shown.

### Describe the behavior you expected

Descriptions for fields should be shown independent of type of validators.

### Steps to reproduce the issue

1. Start up denonavr configuration flow.
2. Host field should have a description of "Leave blank to use auto-discovery"

### What version of Home Assistant Core has the issue?

2022.10.5

### What was the last working version of Home Assistant Core?

_No response_

### In which browser are you experiencing the issue with?

_No response_

### Which operating system are you using to run this browser?

_No response_

### State of relevant entities

_No response_

### Problem-relevant frontend configuration

_No response_

### Javascript errors shown in your browser console/inspector

_No response_

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by steverep**
## Proposed change

`ha-form-string` will render a helper, but it is not passed by `ha-form`.  This fixes that.  Float and integer do not currently render a helper if passed.  I'm not sure if that was on purpose.

## Type of change

- [ ] Dependency upgrade
- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #14164 
- This PR is related to issue or discussion:
- Link to documentation pull request:

## Checklist

- [x] The code change is tested and works locally.
- [x] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

### CODE REVIEW

**Comment by steverep** [comment]
Seemed like the right thing to do so I added it to the others.

File: src/components/ha-form/ha-form-float.ts [hunk 1]
```diff
@@ -29,10 +31,12 @@ export class HaFormFloat extends LitElement implements HaFormElement {
...
         inputMode="decimal"
         .label=${this.label}
+        .helper=${this.helper}
+        helperPersistent
         .value=${this.data !== undefined ? this.data : ""}
         .disabled=${this.disabled}
-        .required=${this.schema.required}
-        .autoValidate=${this.schema.required}
+        ?required=${this.schema.required}
+        ?autoValidate=${this.schema.required}
```
Comment (bramkragten) [inline src/components/ha-form/ha-form-float.ts:39]: Why change it to an attribute?

File: src/components/ha-form/ha-form.ts [hunk 2]
```diff
@@ -92,6 +92,7 @@ export class HaForm extends LitElement implements HaFormElement {
                   schema: item,
                   data: getValue(this.data, item),
                   label: this._computeLabel(item, this.data),
+                  helper: this._computeHelper(item),
                   disabled: this.disabled,
                   hass: this.hass,
                   computeLabel: this.computeLabel,
```
Comment (bramkragten) [inline src/components/ha-form/ha-form.ts:98]: Why do we pass these functions along?

File: src/components/ha-form/ha-form-float.ts (hunk 1 above)
Comment (steverep) [reply src/components/ha-form/ha-form-float.ts:39]: Fixing incompatible types

File: src/components/ha-form/ha-form.ts (hunk 2 above)
Comment (steverep) [reply src/components/ha-form/ha-form.ts:98]: I didn't touch that code.  I believe they are needed by the grid schema.

Comment (bramkragten) [reply src/components/ha-form/ha-form.ts:98]: oh right

**Review by piitaya — DISMISSED** [review]

File: src/components/ha-form/ha-form-float.ts (hunk 1 above)
Comment (steverep) [reply src/components/ha-form/ha-form-float.ts]: I just undid those changes as unrelated and there are a number of other type incompatibilities in those files

**Review by bramkragten — APPROVED** [review]

**Comment by elupus** [comment]
Any chance this could be fixed for checkbox/boolean types as well?

**Comment by steverep** [comment]
Yes, but not as straightforward.  The boolean for `ha-form` is currently rendered quite differently from the boolean selector.  The boolean selector does render a helper, but it is not programmatically associated with the field as it should be for accessibility.

They should be aligned in style and used the same code as part of the fix IMO, and the a11y issue should be incorporated as well.

Can you open an issue for this and tag me?

**Pull request merged**